Marlin/Marlin_main.cpp

```diff
@@ -1649,14 +1650,14 @@ static void do_blocking_move_to(float x, float y, float z, float feed_rate = 0.0
     if (DEBUGGING(LEVELING)) print_xyz(PSTR("do_blocking_move_to"), NULL, x, y, z);
   #endif
 
+  destination[X_AXIS] = x;
+  destination[Y_AXIS] = y;
+  destination[Z_AXIS] = z;
+
   #if ENABLED(DELTA)
 
     feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
 
-    destination[X_AXIS] = x;
-    destination[Y_AXIS] = y;
-    destination[Z_AXIS] = z;
```

Comment:
This is correct. The `do_blocking_move_to` function updates `current_position`, not `destination`. The value of `destination` is scratch for this function, and we should expect `destination` to be trashed by any of the `blocking_move` functions.